File: solution/go/771.go
// Use hashmap to speed up the searching process
func numJewelsInStones(J string, S string) int {
    jewels := make(map[rune]bool)
    for _, val := range J {
        if _, ok := jewels[val]; !ok {
            jewels[val] = true
        }
    }
    count := 0
    for _, val := range S {
        if _, ok := jewels[val]; ok {
            count++
        }
    }
    return count
}

File: solution/python/771.py
"""
Simple question, use counter or `in` operation
"""
class Solution(object):
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        from collections import Counter
        jewels, stones = set(J), Counter(S)
        count = 0
        for key in jewels:
            count += stones[key]
        return count
